src/main/java/com/evilcorp/bloom/model/Customer.java

```diff
@@ -0,0 +1,48 @@
+package com.evilcorp.bloom.model;
+
+import java.util.UUID;
+
+import org.springframework.data.annotation.Id;
+import org.springframework.data.relational.core.mapping.Table;
+
+@Table("customers")
+public class Customer {
+  @Id
+  private UUID id;
```

Comment:
wow, so, in fact, it [does not guarantee no conflicts,](https://dba.stackexchange.com/questions/214083/what-happens-when-an-automatically-generated-uuid-primary-key-collides-in-postgr/214089#214089) albeit the chances of it happening are very low. 

The problem is that there are no good solutions for this. I could use uuid v1 to guarantee no collisions, but then we have privacy/security concerns (this version can reveal information about the system generating the UUID).

From my understanding, if there is a collision, the creation of the entity would fail, which I think is better than decreasing security in the system. So I'd leave it like this.

@shankarnakai what do you suggest

Reply:
discussed this topic in person

- will keep uuidv4
- will handle the collision scenario differently in the application. Include one retry to create the customer. In real life, this might be overkill but since the goal is to practice, will add this as a issue. (: